### DISCUSSION TITLE
Ultima 17.0.0: p-checkbox has a white background when checked in "Input Style = filled" mode

### DISCUSSION THREAD

**Discussion by ThorstenKunz** [primefaces/community#1006] [PrimeNG Templates]
You can reproduce it with your own example page. Just go to https://www.primefaces.org/ultima-ng/#/uikit/input and in the configuration menu switch "Input Style" from "Outlined" to "Filled". Then click on any checkbox. It will actually perform selection but the background of the checkbox will remain white and with the actualy check symbok also being wite it looks like the box is not checked.

**Comment by cetincakiroglu** [comment]
Hi,

Thanks a lot for reporting the issue, we'll investigate it.

**Reply by MWHarrington** [reply]
I recently migrated to Ultima 17.0.0 and noticed the styling of the border color for p-checkbox doesn't take the theme color, it just stays grey. I know that it did take the theme color when I was on 15 (migrate 2 versions one after the other). I've also noticed that this affects the invalid state not showing red as well. This can be seen in your example as mentioned in the initial comment. To see the invalid state just add the classes ng-dirty and ng-invalid to the p-checkbox element using the dev tools. Seems like something changed for the styling ordering/precedence.

Screenshot showing how the styling overridding appears after checking the checkbox and adding the ng-dirty and ng-invalid classes.
![image](https://github.com/primefaces/community/assets/11353672/6a54a579-6b09-4cf3-bed1-642b68605abc)

**Reply by cetincakiroglu** [reply]
Hi,

Thanks for the explanation, ng-invalid and ng-dirty styles shouldn't be overridden by other class. I'm labeling the issue as `bug` and we'll update the template soon.
